rpcv02/chain.go

```diff
@@ -22,11 +23,18 @@ func (provider *Provider) ChainID(ctx context.Context) (string, error) {
 
 // Syncing checks the syncing status of the node.
 func (provider *Provider) Syncing(ctx context.Context) (*SyncStatus, error) {
-	// TODO: manage the fact SyncingStatus is set to false.
-	var result SyncStatus
+	var result interface{}
 	// Note: []interface{}{}...force an empty `params[]` in the jsonrpc request
 	if err := provider.c.CallContext(ctx, &result, "starknet_syncing", []interface{}{}...); err != nil {
 		return nil, err
 	}
-	return &result, nil
+	switch res := result.(type) {
+	case bool:
+		return &SyncStatus{SyncStatus: res}, nil
```

Comment:
Not sure if this falls under rpc v0.2 scope but the only possible value for `result` when it has bool type is `False`. Should we check that here?

Reply:
I don't think we need to as the endpoint will (/should) never return True